lib/liquid/tags/case.rb

```diff
@@ -20,7 +20,11 @@ def initialize(tag_name, markup, options)
 
     def parse(tokens)
       body = new_body
-      body = @blocks.last.attachment while parse_body(body, tokens)
+      while parse_body(body, tokens)
+        body.freeze(parse_context)
```

Comment:
Block body freezing seems like something that should be handled in Liquid::Block itself, such that it doesn't require changes to these individual block tags.

These block tag classes that call `parse_body` all just freeze the body immediately afterwards, so why don't we have `parse_body` freeze the block body by default.  The one use case for partially parsing the body seems to be in Liquid::Block#parse itself, where it could call pass a `partial: true` keyword argument to avoid freezing the block body, then can do the manual `@body.freeze` at the end of Liquid::Block#parse that you are already doing.

Reply:
The original implementation (the one you just reviewed), the `BlockBody#freeze` method was not a true freeze, so `BlockBody#remove_blank_strings` was allowed to run even after freezing (this is not a problem in Shopify/liquid-c#86 because it replaces the strings with a jump instruction, which does not change the size of any data structures). However, now that `BlockBody#freeze` really does freeze the object, we can no longer call `BlockBody#remove_blank_strings` afterwards. So I've changed `Block#parse_body` to have an additional side-effect of calling `BlockBody#remove_blank_strings` before it calls `BlockBody#freeze`. I'm not 100% sure this is fine to do.